src/core/src/pass/serialize.cpp

```diff
@@ -1012,7 +1012,8 @@ void serializeFunc(std::ostream& xml_file,
 }  // namespace
 
 namespace ov {
-bool pass::Serialize::run_on_model(const std::shared_ptr<ngraph::Function>& f) {
+bool pass::Serialize::run_on_model(const std::shared_ptr<ngraph::Function>& f_orig) {
+    auto f = ngraph::clone_function(*f_orig);
```

Comment:
Add a comment with explanations why this copy is needed here.

Reply:
done